### DISCUSSION TITLE
Create a `useAnime` Composable

### DISCUSSION THREAD

**Discussion by BayBreezy** [ivodolenc/nuxt-animejs#12] [Ideas]
### Request Description

Hello, 

Instead of having to extract the `$anime` function from the nuxtApp, why not add a `useAnime` composable that can be used right away?

It can be added in the `module.ts` file.

### Alternative Solutions

NA

**Comment by ivodolenc** [comment]
Hi, thanks for the suggestion.

Nice idea, but maybe an opt-in option should be added:

```ts
// nuxt.config.ts

{
  anime: {
    composables: true // default is undefined
  }
}
```

With this option, the user chooses which method he wants to use. If enables composables, then `provide` ($anime) is disabled and vice versa.

Main reason is to avoid multiple global injections at the same time (provide and composable).

Also, keep in mind that the user can disable built-in [auto imports](https://nuxt.com/docs/guide/concepts/auto-imports#disable-auto-imports).

In that case we will need an explicit import:

```ts
// Explicit import (optional)
import { useAnime } from '#anime'
```

and the additional `autoImport` module option:

```ts
{
  anime: {
    autoImport: true // default is true
  }
}
```

By default, module `autoImport` are enabled.

---

I'm interested in merging this, I think it would be nice addition to module.

Can you please update the PR? Don't bother with the docs and examples, I can edit it later, no problem.

**Reply by BayBreezy** [reply]
Ok no problem. I will Update it to be an opt-in feature and let you know.

**Comment by BayBreezy** [comment]
@ivodolenc See here for the changes made --> https://github.com/hypernym-studio/nuxt-anime/pull/13/commits/f910e35e3875234a0f3963424d051b5272267d75.

Now if yo pass this to your `nuxt.config`, the composable will be added, if not, then the plugin is added

```ts
export default defineNuxtConfig({
// ...
  anime:{
    composable: true
  }
// ...
})
```

**Comment by ivodolenc** [comment]
This feature is now available in `2.1.0` version.

To use `useAnime` custom composable globally, enable it via module options:

```ts
// nuxt.config.ts

{
  anime: {
    composables: true
  }
}
```

Additionally, if you prefer to use only composables, you can disable the `$anime` helper:

```diff
// nuxt.config.ts

{
  anime: {
+   provide: false,
    composables: true
  }
}
```

See the [docs](https://github.com/hypernym-studio/nuxt-anime#options) for more info.

@BayBreezy Thanks again for your time and contribution.
